### PR TITLE
Add strongly typed options to Intl modules

### PR DESCRIPTION
**Pull request by glennsl**

### CODE REVIEW

**Comment by zth** [comment]
Good stuff @glennsl ! Ready to merge?

**Comment by glennsl** [comment]
As far as I'm aware, yes. I wouldn't count out having made some mistakes though, so a thorough review wouldn't be a bad idea.

**Comment by cknitt** [comment]
@glennsl Thanks!

There is also #65 (a more encompassing Intl bindings PR). One difference I see is that #65 distinguishes between `Intl.Collator.options`, `Intl.Collator.resolvedOptions` and `Intl.Collator.supportedLocalesOptions`.

Also see the discussion there starting with https://github.com/rescript-association/rescript-core/pull/65#issuecomment-1872567799 (polymorphic variants vs. regular variants).

File: src/intl/Core__Intl__Collator.res [hunk 1]
```diff
@@ -1,17 +1,27 @@
 type t
 
+type options = {
+  usage?: [#sort | #search],
+  localeMathcer?: [#lookup | #"best fit"],
```
Comment (cknitt) [inline src/intl/Core__Intl__Collator.res:5]: Typo (occurs several times)

**Comment by glennsl** [comment]
> There is also #65 (a more encompassing Intl bindings PR). One difference I see is that #65 distinguishes between `Intl.Collator.options`, `Intl.Collator.resolvedOptions` and `Intl.Collator.supportedLocalesOptions`.
> 
> Also see the discussion there starting with [#65 (comment)](https://github.com/rescript-association/rescript-core/pull/65#issuecomment-1872567799) (polymorphic variants vs. regular variants).

Ah, I've missed that. That's so much better! This is mostly just a lazy copy of my own bindings with some additions to flesh it out. I'll just close this and comment over there then.

**Pull request closed without merging**